pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -326,11 +326,31 @@ func (c *Cluster) postReconcileUpdateOSDProperties(desiredOSDs map[int]*OSDInfo)
 	}
 	logger.Debugf("post processing osd properties with %d actual osds from ceph osd df and %d existing osds found during reconcile", len(osdUsage.OSDNodes), len(desiredOSDs))
 	for _, actualOSD := range osdUsage.OSDNodes {
-		if desiredOSD, ok := desiredOSDs[actualOSD.ID]; ok {
-			if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
-				// Log the error and allow other updates to continue
-				logger.Error(err)
-			}
+		desiredOSD, ok := desiredOSDs[actualOSD.ID]
+		if !ok {
+			return nil
+		}
+		if err := c.resizeOsdCrushWeight(actualOSD); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error(err)
+		}
+		if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error(err)
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) resizeOsdCrushWeight(actualOSD cephclient.OSDNodeUsage) error {
+	if !c.spec.Storage.ResizeOsdCrushWeight {
+		return nil
+	}
+	if actualOSD.KB != actualOSD.CrushWeight {
```

Comment:
Shouldn't `actualOSD.KB` be translated into a TiB value? Or is it TB? We should make sure we know which should be used.

Reply:
Its acutally KIb and we need to convert to Tebibyte